```{r, include = FALSE}
ottrpal::set_knitr_image_path()
```

# SLiM Guide

Notes on usage of SLiM for evolutionary modeling. Adapted from the SLiM manual: http://benhaller.com/slim/SLiM_Manual.pdf

## A Basic SLiM Simulation

Our first simulation will be a lightly modified version of the script that comes up when we first open the SLiM graphical user interface (GUI). It is a simple simulation tracking one population, with no selection:

```{r, engine='bash', eval=FALSE}
// set up a simple neutral simulation
initialize() {
	initializeMutationRate(1e-7);
	
	// m1 mutation type: neutral
	initializeMutationType("m1", 0.5, "f", 0.0);
	
	// g1 genomic element type: uses m1 for all mutations
	initializeGenomicElementType("g1", m1, 1.0);
	
	// uniform chromosome of length 100 kb with uniform recombination
	initializeGenomicElement(g1, 0, 99999);
	initializeRecombinationRate(1e-8);
}

// create a population of 500 individuals
1 early() {
	sim.addSubpop("p1", 500);
}

2000 late() { 
  sim.outputFixedMutations(); 
}
```

## Initializing a Simulation 

The first step in running a simulation is defining all of the parameters that characterize our populations - how many populations are we studying? How big are they? What do individual genomes look like? We use an `initialize()` block to define any parameters like this. 

In almost every simulation, we will define a core set of common parameters. They are: 

### Global Mutation Rates

The mutation rate is set by the command `initializeMutationRate()`. In its simplest form, we run this function with a single argument, the mutation, as such: `initializeMutationRate(1e-7);`. Now, when executing the simulation, SLiM will go through each gamete base by base, and will introduce a mutation with a probability of `1e-7`.

### Mutation Types

When running our simulation, we can distinguish between multiple kinds of mutations, each with their own prevalence and impact - for example, a neutral substitution, or a lethal deletion, or a rare variant that confers a selective advantage. To create a mutation type, we use the `initializeMutationType()` command. For example, this following line of code creates a deleterious mutation: 

```{r, engine='python', eval=FALSE}
initializeMutationType("m1", 0.5, "f", -0.02); 
```

Let's break the arguments to `initializeMutationType` down: 

* the **mutation id**: this can be any integer, or a string in the format `m<integer>`. This is the name that you will use to keep track of your mutation.
* the **dominance coefficient**: Here, `0.5`. This is used when determining how a mutation impacts the fitness of your individual. A mutation with a dominance coefficient of `1.0` is completely dominant; a mutation with a dominance coefficient of `0.0` is recessive. Mutations with values in between represent incomplete dominance - in our case, `0.5` indicates that heterozygotes have half the fitness effect of the mutation
* the last two arguments, `"f"` and `-0.02` denote the **distribution** and **magnitude** of **how our mutation impacts fitness**. `f` indicates that the effect is *fixed* - that is, the mutation impacts all affected individuals identically. The value `-0.02` indicates that this effect is a fitness reduction of 0.02 - any individual homozygous for this allele is expected to have offspring at 0.98 the frequency of an individual without this allele, all else being equal. While this allele has the same effect on all carriers, we can also create alleles with different fitness effect distributions - for example `"n"` indicates that for each individual with this mutation, we draw the fitness effect from a normal distribution, `"e"` indicates that we draw fitness effects from the exponential distribution. For some of these, e.g. the normal distribution, we may need to provide more than one numerical parameter.  

### Types of DNA 

Now that we have defined the types of mutations we can encounter in our simulation, we can no specify the types of genomic regions that we are working with. For example, we can create genomic regions that represent telomeres, centromeres, introns, exons, etc. And intuitively, we can anticipate that mutations are likely to have different effects in different regions (for example, a SNP in an exon is more likely to have a dramatic phenotypic impact than a SNP in a centromere). 

We can define a type of DNA using the `initializeGenomicElementType()` function. This function takes three argument, in the following order: 

* the **name** of the genomic region (in the form `"g"` + an integer)
* the **kinds of mutations** that can occur in this type of DNA region. 
* the **relative frequencies** of different mutation types

Here is a simple example of this command: 

`initializeGenomicElementType("g1", m1, 1.0);`

This creates a type of DNA region called `"g1"`, which can only have `m1` type mutations. Since there is only one type of mutation, it happens at a frequency of `1`.

### Genomic Regions 

Now that we have defined a type of DNA that we can see in our genome, we need to great a specific instance of it. This is done with the `initializeGenomicElement()` command. This takes as arguments:

* the *type of genomic element* 
* the *start coordinate* 
* the *stop coordinate*

For example:

`initializeGenomicElement(g1, 0, 99999);` 

This creates a genomic element of type `g1`, which extends from bases 0 to 99999.

#### Recombination Rates 

Lastly, we need to set the recombination rate. This is done using the `initializeRecombinationRate()` function, which functions similarly to how we defined mutation rates. For this function, we just provide a float defining the per base recombination rate: 

`initializeRecombinationRate(1e-8);`
