<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3.8 Introducing Sweep Mutations | Syllabus</title>
  <meta name="description" content="Description about Course/Book." />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="3.8 Introducing Sweep Mutations | Syllabus" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Description about Course/Book." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3.8 Introducing Sweep Mutations | Syllabus" />
  
  <meta name="twitter:description" content="Description about Course/Book." />
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="assets/peppa.png" type="image/x-icon" />
<link rel="prev" href="storing-information.html"/>
<link rel="next" href="writing-output.html"/>
<script src="libs/header-attrs-2.10/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="assets/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<a href="https://andrew-bortvin.github.io/slimNotes/"><img src="assets/peppa.png" style=" width: 80%; padding-left: 40px; padding-top: 8px; vertical-align: top "</a>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Population Genetics: Simulation and Visualization</a></li>
<li class="chapter" data-level="2" data-path="syllabus-population-genetics-simulation-visualization.html"><a href="syllabus-population-genetics-simulation-visualization.html"><i class="fa fa-check"></i><b>2</b> Syllabus: Population Genetics Simulation &amp; Visualization</a>
<ul>
<li class="chapter" data-level="2.0.1" data-path="syllabus-population-genetics-simulation-visualization.html"><a href="syllabus-population-genetics-simulation-visualization.html#course-description"><i class="fa fa-check"></i><b>2.0.1</b> Course Description</a></li>
<li class="chapter" data-level="2.0.2" data-path="syllabus-population-genetics-simulation-visualization.html"><a href="syllabus-population-genetics-simulation-visualization.html#class-structure"><i class="fa fa-check"></i><b>2.0.2</b> Class Structure</a></li>
<li class="chapter" data-level="2.0.3" data-path="syllabus-population-genetics-simulation-visualization.html"><a href="syllabus-population-genetics-simulation-visualization.html#schedule"><i class="fa fa-check"></i><b>2.0.3</b> Schedule</a></li>
<li class="chapter" data-level="2.0.4" data-path="syllabus-population-genetics-simulation-visualization.html"><a href="syllabus-population-genetics-simulation-visualization.html#grading"><i class="fa fa-check"></i><b>2.0.4</b> Grading</a></li>
<li class="chapter" data-level="2.0.5" data-path="syllabus-population-genetics-simulation-visualization.html"><a href="syllabus-population-genetics-simulation-visualization.html#student-hours"><i class="fa fa-check"></i><b>2.0.5</b> Student Hours</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="slim-guide.html"><a href="slim-guide.html"><i class="fa fa-check"></i><b>3</b> SLiM Guide</a>
<ul>
<li class="chapter" data-level="3.1" data-path="a-basic-slim-simulation.html"><a href="a-basic-slim-simulation.html"><i class="fa fa-check"></i><b>3.1</b> A Basic SLiM Simulation</a></li>
<li class="chapter" data-level="3.2" data-path="initliazing-a-simulation.html"><a href="initliazing-a-simulation.html"><i class="fa fa-check"></i><b>3.2</b> Initliazing a Simulation</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="initliazing-a-simulation.html"><a href="initliazing-a-simulation.html#global-mutation-rates"><i class="fa fa-check"></i><b>3.2.1</b> Global Mutation Rates</a></li>
<li class="chapter" data-level="3.2.2" data-path="initliazing-a-simulation.html"><a href="initliazing-a-simulation.html#mutation-types"><i class="fa fa-check"></i><b>3.2.2</b> Mutation Types</a></li>
<li class="chapter" data-level="3.2.3" data-path="initliazing-a-simulation.html"><a href="initliazing-a-simulation.html#types-of-dna"><i class="fa fa-check"></i><b>3.2.3</b> Types of DNA</a></li>
<li class="chapter" data-level="3.2.4" data-path="initliazing-a-simulation.html"><a href="initliazing-a-simulation.html#genomic-regions"><i class="fa fa-check"></i><b>3.2.4</b> Genomic Regions</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="running-a-simulation.html"><a href="running-a-simulation.html"><i class="fa fa-check"></i><b>3.3</b> Running a simulation</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="running-a-simulation.html"><a href="running-a-simulation.html#adding-subpopulations"><i class="fa fa-check"></i><b>3.3.1</b> Adding subpopulations</a></li>
<li class="chapter" data-level="3.3.2" data-path="running-a-simulation.html"><a href="running-a-simulation.html#events-that-occur-every-generation"><i class="fa fa-check"></i><b>3.3.2</b> Events that occur every generation</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="ending-a-simulation.html"><a href="ending-a-simulation.html"><i class="fa fa-check"></i><b>3.4</b> Ending a simulation</a></li>
<li class="chapter" data-level="3.5" data-path="multiple-mutations-elements-and-regions.html"><a href="multiple-mutations-elements-and-regions.html"><i class="fa fa-check"></i><b>3.5</b> Multiple Mutations, Elements and Regions</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="multiple-mutations-elements-and-regions.html"><a href="multiple-mutations-elements-and-regions.html#making-multiple-mutations"><i class="fa fa-check"></i><b>3.5.1</b> Making multiple mutations</a></li>
<li class="chapter" data-level="3.5.2" data-path="multiple-mutations-elements-and-regions.html"><a href="multiple-mutations-elements-and-regions.html#genomic-elements-with-multiple-mutations"><i class="fa fa-check"></i><b>3.5.2</b> Genomic Elements with multiple mutations</a></li>
<li class="chapter" data-level="3.5.3" data-path="multiple-mutations-elements-and-regions.html"><a href="multiple-mutations-elements-and-regions.html#multiple-genomic-elements-types"><i class="fa fa-check"></i><b>3.5.3</b> Multiple Genomic Elements Types</a></li>
<li class="chapter" data-level="3.5.4" data-path="multiple-mutations-elements-and-regions.html"><a href="multiple-mutations-elements-and-regions.html#multiple-genomic-elements"><i class="fa fa-check"></i><b>3.5.4</b> Multiple Genomic Elements</a></li>
<li class="chapter" data-level="3.5.5" data-path="multiple-mutations-elements-and-regions.html"><a href="multiple-mutations-elements-and-regions.html#slim-gui"><i class="fa fa-check"></i><b>3.5.5</b> SLiM GUI</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="week-1-assignment.html"><a href="week-1-assignment.html"><i class="fa fa-check"></i><b>3.6</b> Week 1 Assignment</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="week-1-assignment.html"><a href="week-1-assignment.html#population-size"><i class="fa fa-check"></i><b>3.6.1</b> 1.1 - Population Size</a></li>
<li class="chapter" data-level="3.6.2" data-path="week-1-assignment.html"><a href="week-1-assignment.html#selection"><i class="fa fa-check"></i><b>3.6.2</b> 1.2 - Selection</a></li>
<li class="chapter" data-level="3.6.3" data-path="week-1-assignment.html"><a href="week-1-assignment.html#multiple-mutation-types-multiple-genomic-regions"><i class="fa fa-check"></i><b>3.6.3</b> 1.3 - Multiple mutation types; multiple genomic regions</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="storing-information.html"><a href="storing-information.html"><i class="fa fa-check"></i><b>3.7</b> Storing Information</a>
<ul>
<li class="chapter" data-level="3.7.1" data-path="storing-information.html"><a href="storing-information.html#local-variables"><i class="fa fa-check"></i><b>3.7.1</b> Local variables</a></li>
<li class="chapter" data-level="3.7.2" data-path="storing-information.html"><a href="storing-information.html#constants"><i class="fa fa-check"></i><b>3.7.2</b> Constants</a></li>
<li class="chapter" data-level="3.7.3" data-path="storing-information.html"><a href="storing-information.html#simulation-values"><i class="fa fa-check"></i><b>3.7.3</b> Simulation Values</a></li>
</ul></li>
<li class="chapter" data-level="3.8" data-path="introducing-sweep-mutations.html"><a href="introducing-sweep-mutations.html"><i class="fa fa-check"></i><b>3.8</b> Introducing Sweep Mutations</a>
<ul>
<li class="chapter" data-level="3.8.1" data-path="introducing-sweep-mutations.html"><a href="introducing-sweep-mutations.html#defining-the-mutation"><i class="fa fa-check"></i><b>3.8.1</b> Defining the Mutation</a></li>
<li class="chapter" data-level="3.8.2" data-path="introducing-sweep-mutations.html"><a href="introducing-sweep-mutations.html#sampling-individuals"><i class="fa fa-check"></i><b>3.8.2</b> Sampling individuals</a></li>
</ul></li>
<li class="chapter" data-level="3.9" data-path="writing-output.html"><a href="writing-output.html"><i class="fa fa-check"></i><b>3.9</b> Writing Output</a></li>
<li class="chapter" data-level="3.10" data-path="code-from-week-2.html"><a href="code-from-week-2.html"><i class="fa fa-check"></i><b>3.10</b> Code from Week 2</a></li>
<li class="chapter" data-level="3.11" data-path="week-2-assignment.html"><a href="week-2-assignment.html"><i class="fa fa-check"></i><b>3.11</b> Week 2 Assignment</a>
<ul>
<li class="chapter" data-level="3.11.1" data-path="week-2-assignment.html"><a href="week-2-assignment.html#hard-sweeps"><i class="fa fa-check"></i><b>3.11.1</b> Hard Sweeps</a></li>
<li class="chapter" data-level="3.11.2" data-path="week-2-assignment.html"><a href="week-2-assignment.html#advanced-exercises"><i class="fa fa-check"></i><b>3.11.2</b> Advanced Exercises</a></li>
<li class="chapter" data-level="3.11.3" data-path="week-2-assignment.html"><a href="week-2-assignment.html#advanced-exercise-1---hard-sweeps-recombination-and-hitchhiker-mutations."><i class="fa fa-check"></i><b>3.11.3</b> Advanced Exercise 1 - Hard Sweeps, Recombination, and Hitchhiker mutations.</a></li>
<li class="chapter" data-level="3.11.4" data-path="week-2-assignment.html"><a href="week-2-assignment.html#advanced-exercise-2---soft-sweeps-and-burts-of-environmental-pressure"><i class="fa fa-check"></i><b>3.11.4</b> Advanced Exercise 2 - Soft Sweeps and Burts of Environmental Pressure</a></li>
</ul></li>
<li class="chapter" data-level="3.12" data-path="multiple-populations.html"><a href="multiple-populations.html"><i class="fa fa-check"></i><b>3.12</b> Multiple Populations</a>
<ul>
<li class="chapter" data-level="3.12.1" data-path="multiple-populations.html"><a href="multiple-populations.html#adding-a-new-population"><i class="fa fa-check"></i><b>3.12.1</b> Adding a new population</a></li>
<li class="chapter" data-level="3.12.2" data-path="multiple-populations.html"><a href="multiple-populations.html#migration-between-populations"><i class="fa fa-check"></i><b>3.12.2</b> Migration between populations</a></li>
<li class="chapter" data-level="3.12.3" data-path="multiple-populations.html"><a href="multiple-populations.html#splitting-a-population-into-two"><i class="fa fa-check"></i><b>3.12.3</b> Splitting a population into two</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="papers-of-interest.html"><a href="papers-of-interest.html"><i class="fa fa-check"></i><b>4</b> Papers of Interest</a>
<ul>
<li class="chapter" data-level="4.0.1" data-path="papers-of-interest.html"><a href="papers-of-interest.html#week-2---selection-and-sweeps"><i class="fa fa-check"></i><b>4.0.1</b> Week 2 - Selection and Sweeps</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="lecture-slides.html"><a href="lecture-slides.html"><i class="fa fa-check"></i><b>5</b> Lecture Slides</a></li>
<li class="chapter" data-level="" data-path="about-the-authors.html"><a href="about-the-authors.html"><i class="fa fa-check"></i>About the Authors</a></li>
<li class="chapter" data-level="6" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>6</b> References</a></li>
<li class="divider"></li>
<p style="text-align:center;"> <a href="https://github.com/jhudsl/OTTR_Template" target="blank" > This content was published with</a> <a href="https://bookdown.org/" target="blank"> bookdown by:</a> </p>
<p style="text-align:center;"> <a href="http://jhudatascience.org/"> The Johns Hopkins Data Science Lab </a></p>
<p style="text-align:center; font-size: 12px;"> <a href="https://github.com/rstudio4edu/rstudio4edu-book/"> Style adapted from: rstudio4edu-book </a> <a href ="https://creativecommons.org/licenses/by/2.0/"> (CC-BY 2.0) </a></p>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Syllabus</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="introducing-sweep-mutations" class="section level2" number="3.8">
<h2><span class="header-section-number">3.8</span> Introducing Sweep Mutations</h2>
<p>This week, we are simulating <strong>selective sweeps</strong>, or events in which a genetic variant rises rapidly in frequency due to positive selection.</p>
<p>We want to mimic a so-called “soft sweep”, or a scenario in which an existing genetic variant becomes favorable (e.g. because of changing environmental pressures).</p>
<p>In practice, a simple way of doing this is to randomly introduce a favorable mutation at a set frequency at a pre-specified generation. This process can be divided into a few sequential steps:</p>
<ol style="list-style-type: decimal">
<li>Define the mutation type for your sweep</li>
<li>At a given generation, sample a predetermined of your individuals</li>
<li>To sampled individuals, add the mutation at a predetermined genomic position</li>
</ol>
<div id="defining-the-mutation" class="section level3" number="3.8.1">
<h3><span class="header-section-number">3.8.1</span> Defining the Mutation</h3>
<p>Let’s start with a familiar <code>initialize</code> statement, where we have one neutral mutation type and one genomic region.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb21-1"><a href="introducing-sweep-mutations.html#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">initialize()</span><span class="kw">{</span></span>
<span id="cb21-2"><a href="introducing-sweep-mutations.html#cb21-2" aria-hidden="true" tabindex="-1"></a>    <span class="ex">initializeMutationRate</span><span class="er">(</span><span class="ex">1e-7</span><span class="kw">);</span></span>
<span id="cb21-3"><a href="introducing-sweep-mutations.html#cb21-3" aria-hidden="true" tabindex="-1"></a>    <span class="ex">//</span> Create mutation types</span>
<span id="cb21-4"><a href="introducing-sweep-mutations.html#cb21-4" aria-hidden="true" tabindex="-1"></a>    <span class="ex">initializeMutationType</span><span class="er">(</span><span class="st">&quot;m1&quot;</span><span class="ex">,</span> 0.5, <span class="st">&quot;f&quot;</span>, 0.0<span class="kw">);</span></span>
<span id="cb21-5"><a href="introducing-sweep-mutations.html#cb21-5" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb21-6"><a href="introducing-sweep-mutations.html#cb21-6" aria-hidden="true" tabindex="-1"></a>    <span class="ex">//</span> Create Types of DNA </span>
<span id="cb21-7"><a href="introducing-sweep-mutations.html#cb21-7" aria-hidden="true" tabindex="-1"></a>    <span class="ex">initializeGenomicElementType</span><span class="er">(</span><span class="st">&quot;g1&quot;</span><span class="ex">,</span> m1, 1<span class="kw">);</span>   </span>
<span id="cb21-8"><a href="introducing-sweep-mutations.html#cb21-8" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb21-9"><a href="introducing-sweep-mutations.html#cb21-9" aria-hidden="true" tabindex="-1"></a>    <span class="ex">//</span> Arrange DNA into a genome</span>
<span id="cb21-10"><a href="introducing-sweep-mutations.html#cb21-10" aria-hidden="true" tabindex="-1"></a>    <span class="ex">initializeGenomicElement</span><span class="er">(</span><span class="ex">g1,</span> 0, 9999<span class="kw">);</span></span>
<span id="cb21-11"><a href="introducing-sweep-mutations.html#cb21-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-12"><a href="introducing-sweep-mutations.html#cb21-12" aria-hidden="true" tabindex="-1"></a>    <span class="ex">initializeRecombinationRate</span><span class="er">(</span><span class="ex">1e-8</span><span class="kw">);</span></span>
<span id="cb21-13"><a href="introducing-sweep-mutations.html#cb21-13" aria-hidden="true" tabindex="-1"></a><span class="kw">}</span></span></code></pre></div>
<p>To introduce our favorable mutation, let’s add the following to our <code>initialize</code> statement:</p>
<p><code>initializeMutationType("m2", 0.5, "f", 0.015);</code>
<code>m2.mutationStackPolicy="l";</code>
<code>m2.convertToSubstitution = F;</code></p>
<p>The first line should be familiar - we have a mutation type called <code>"m2"</code>, which has a positive fitness effect of 0.015.</p>
<p>The second line, <code>m2.mutationStackPolicy="l";</code> controls what happens when multiple mutations occur at the same position. By default, SLiM allows multiple mutations at the same position, with additive fitness effects. Since we are modeling a single base substitution, we want to only track the most recent mutation at a given position. setting the mutation stack policy to <code>"l"</code> tells SLiM to only care about the last mutation at a given site.</p>
<p>The final line, <code>m2.convertToSubstitution = F;</code>, controls how our mutation behaves when it reaches fixation. By default, once a mutation reaches fixation, it is converted to a <strong>substitution</strong> - that is, SLiM stops tracking the mutation and only keeps a log of the coordinate where the mutation happened. This is because once the variant is fixed, it has the same impact on all individuals. Here, we want to disable this behavior, since we want to track this specific variant throughout the entire simulation.</p>
</div>
<div id="sampling-individuals" class="section level3" number="3.8.2">
<h3><span class="header-section-number">3.8.2</span> Sampling individuals</h3>
<p>Let’s say that we want to introduce this mutation into our population with an allele frequency of 5%, at position 1000. The following code block does this.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb22-1"><a href="introducing-sweep-mutations.html#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="ex">400</span> late<span class="er">(</span><span class="kw">){</span></span>
<span id="cb22-2"><a href="introducing-sweep-mutations.html#cb22-2" aria-hidden="true" tabindex="-1"></a>    <span class="ex">p1_size</span> = size<span class="er">(</span><span class="ex">p1.genomes</span><span class="kw">);</span>  <span class="ex">//</span> Count how many chromosomes we have </span>
<span id="cb22-3"><a href="introducing-sweep-mutations.html#cb22-3" aria-hidden="true" tabindex="-1"></a>    <span class="ex">initial_freq</span> = asInteger<span class="er">(</span><span class="ex">p1_size</span> <span class="pp">*</span> 0.05<span class="kw">);</span>  <span class="ex">//</span> Find 5% of total number of chromosomes</span>
<span id="cb22-4"><a href="introducing-sweep-mutations.html#cb22-4" aria-hidden="true" tabindex="-1"></a>    <span class="ex">target</span> = sample<span class="er">(</span><span class="ex">p1.genomes,</span> initial_freq<span class="kw">);</span>  <span class="ex">//</span> Randomly sample 5% of chromosomes</span>
<span id="cb22-5"><a href="introducing-sweep-mutations.html#cb22-5" aria-hidden="true" tabindex="-1"></a>    <span class="ex">target.addNewDrawnMutation</span><span class="er">(</span><span class="ex">m2,</span> 1000<span class="kw">);</span>   <span class="ex">//</span> Add new mutation</span>
<span id="cb22-6"><a href="introducing-sweep-mutations.html#cb22-6" aria-hidden="true" tabindex="-1"></a><span class="kw">}</span></span></code></pre></div>
<p>Let’s break this down line by line:</p>
<ul>
<li><code>p1_size = size(p1.genomes);</code>: <code>p1.genomes</code> is a list of all the chromosomes in our population. The <code>size()</code> function gets the length of this list, i.e. the total number of chromosomes. We save this as a local variable, <code>p1_size</code>.</li>
<li><code>initial_freq = asInteger(p1_size * 0.05);</code>. We multiply our number of chromosomes by 0.05 to get the number of chromosomes into which we want to insert the mutation. Depeding on our population size, we might get a non-integer number. Since we can’t add a mutation into, say, 3.2 chromosomes, we convert to an integer, rounding down, using the <code>asInteger()</code> function.</li>
<li><code>target = sample(p1.genomes, initial_freq);</code> The <code>sample()</code> function randomly samples from a set of items. It takes two arguments - what we are sampling from and how many items to sample. This essentially says, we should sample 5% of our genomes.</li>
<li><code>target.addNewDrawnMutation(m2, 1000);</code>: The <code>addNewDrawnMutation()</code> function takes two arguments, the mutation type to add, and the genomic position to add it to. Note that we call the function as <code>target.addNewDrawnMutation()</code> - the <code>target</code> here indicates the object we are operating on.</li>
</ul>
<p>All together, our simulation may look like this:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb23-1"><a href="introducing-sweep-mutations.html#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">initialize()</span><span class="kw">{</span></span>
<span id="cb23-2"><a href="introducing-sweep-mutations.html#cb23-2" aria-hidden="true" tabindex="-1"></a>    <span class="ex">initializeMutationRate</span><span class="er">(</span><span class="ex">1e-7</span><span class="kw">);</span></span>
<span id="cb23-3"><a href="introducing-sweep-mutations.html#cb23-3" aria-hidden="true" tabindex="-1"></a>    <span class="ex">//</span> Create mutation types</span>
<span id="cb23-4"><a href="introducing-sweep-mutations.html#cb23-4" aria-hidden="true" tabindex="-1"></a>    <span class="ex">initializeMutationType</span><span class="er">(</span><span class="st">&quot;m1&quot;</span><span class="ex">,</span> 0.5, <span class="st">&quot;f&quot;</span>, 0.0<span class="kw">);</span></span>
<span id="cb23-5"><a href="introducing-sweep-mutations.html#cb23-5" aria-hidden="true" tabindex="-1"></a>    <span class="ex">initializeMutationType</span><span class="er">(</span><span class="st">&quot;m2&quot;</span><span class="ex">,</span> 1, <span class="st">&quot;f&quot;</span>, 0.015<span class="kw">);</span></span>
<span id="cb23-6"><a href="introducing-sweep-mutations.html#cb23-6" aria-hidden="true" tabindex="-1"></a>    <span class="ex">m2.mutationStackPolicy=</span><span class="st">&quot;l&quot;</span><span class="kw">;</span></span>
<span id="cb23-7"><a href="introducing-sweep-mutations.html#cb23-7" aria-hidden="true" tabindex="-1"></a>    <span class="ex">m2.convertToSubstitution</span> = F<span class="kw">;</span></span>
<span id="cb23-8"><a href="introducing-sweep-mutations.html#cb23-8" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb23-9"><a href="introducing-sweep-mutations.html#cb23-9" aria-hidden="true" tabindex="-1"></a>    <span class="ex">//</span> Create Types of DNA </span>
<span id="cb23-10"><a href="introducing-sweep-mutations.html#cb23-10" aria-hidden="true" tabindex="-1"></a>    <span class="ex">initializeGenomicElementType</span><span class="er">(</span><span class="st">&quot;g1&quot;</span><span class="ex">,</span> m1, 1<span class="kw">);</span>   </span>
<span id="cb23-11"><a href="introducing-sweep-mutations.html#cb23-11" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb23-12"><a href="introducing-sweep-mutations.html#cb23-12" aria-hidden="true" tabindex="-1"></a>    <span class="ex">//</span> Arrange DNA into a genome</span>
<span id="cb23-13"><a href="introducing-sweep-mutations.html#cb23-13" aria-hidden="true" tabindex="-1"></a>    <span class="ex">initializeGenomicElement</span><span class="er">(</span><span class="ex">g1,</span> 0, 9999<span class="kw">);</span></span>
<span id="cb23-14"><a href="introducing-sweep-mutations.html#cb23-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-15"><a href="introducing-sweep-mutations.html#cb23-15" aria-hidden="true" tabindex="-1"></a>    <span class="ex">initializeRecombinationRate</span><span class="er">(</span><span class="ex">1e-8</span><span class="kw">);</span></span>
<span id="cb23-16"><a href="introducing-sweep-mutations.html#cb23-16" aria-hidden="true" tabindex="-1"></a><span class="kw">}</span></span>
<span id="cb23-17"><a href="introducing-sweep-mutations.html#cb23-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-18"><a href="introducing-sweep-mutations.html#cb23-18" aria-hidden="true" tabindex="-1"></a><span class="ex">1</span> early<span class="er">(</span><span class="kw">){</span></span>
<span id="cb23-19"><a href="introducing-sweep-mutations.html#cb23-19" aria-hidden="true" tabindex="-1"></a>    <span class="ex">sim.addSubpop</span><span class="er">(</span><span class="st">&quot;p1&quot;</span><span class="ex">,</span> 500<span class="kw">);</span></span>
<span id="cb23-20"><a href="introducing-sweep-mutations.html#cb23-20" aria-hidden="true" tabindex="-1"></a><span class="kw">}</span></span>
<span id="cb23-21"><a href="introducing-sweep-mutations.html#cb23-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-22"><a href="introducing-sweep-mutations.html#cb23-22" aria-hidden="true" tabindex="-1"></a><span class="ex">400</span> late<span class="er">(</span><span class="kw">){</span></span>
<span id="cb23-23"><a href="introducing-sweep-mutations.html#cb23-23" aria-hidden="true" tabindex="-1"></a>    <span class="ex">p1_size</span> = size<span class="er">(</span><span class="ex">p1.genomes</span><span class="kw">);</span>  <span class="ex">//</span> Count how many chromosomes we have </span>
<span id="cb23-24"><a href="introducing-sweep-mutations.html#cb23-24" aria-hidden="true" tabindex="-1"></a>    <span class="ex">initial_freq</span> = asInteger<span class="er">(</span><span class="ex">p1_size</span> <span class="pp">*</span> 0.05<span class="kw">);</span>  <span class="ex">//</span> Find 5% of total number of chromosomes</span>
<span id="cb23-25"><a href="introducing-sweep-mutations.html#cb23-25" aria-hidden="true" tabindex="-1"></a>    <span class="ex">target</span> = sample<span class="er">(</span><span class="ex">p1.genomes,</span> initial_freq<span class="kw">);</span>  <span class="ex">//</span> Randomly sample 5% of chromosomes</span>
<span id="cb23-26"><a href="introducing-sweep-mutations.html#cb23-26" aria-hidden="true" tabindex="-1"></a>    <span class="ex">target.addNewDrawnMutation</span><span class="er">(</span><span class="ex">m2,</span> 1000<span class="kw">);</span>   <span class="ex">//</span> Add new mutation</span>
<span id="cb23-27"><a href="introducing-sweep-mutations.html#cb23-27" aria-hidden="true" tabindex="-1"></a><span class="kw">}</span></span>
<span id="cb23-28"><a href="introducing-sweep-mutations.html#cb23-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-29"><a href="introducing-sweep-mutations.html#cb23-29" aria-hidden="true" tabindex="-1"></a><span class="ex">2000</span> late<span class="er">(</span><span class="kw">)</span> <span class="kw">{</span> <span class="fu">sim.outputFixedMutations()</span><span class="kw">;</span> <span class="kw">}</span></span></code></pre></div>

</div>
</div>
<hr>
<center> 
  <div class="footer">
      All illustrations <a href="https://creativecommons.org/licenses/by/4.0/">CC-BY. </a>
      <br>
      All other materials <a href= "https://creativecommons.org/licenses/by/4.0/"> CC-BY </a> unless noted otherwise.
  </div>
</center>
            </section>

          </div>
        </div>
      </div>
<a href="storing-information.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="writing-output.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section",
"depth": 1,
"split_bib": 1
}
});
});
</script>

</body>

</html>
